### ISSUE TITLE
Discriminator training and loss

### ISSUE THREAD

**Issue by caaxgave** [LZ-CH/Exposure_Correction-pytorch#7]
Hello! 

Thanks for your approach, in advance. I have some questions about your implementation.
1) Why are you training the discriminator since the very beginning? In MATLAB code it seems to begin discriminator training from epoch 15 on 256x256.
2) Have you trying to use nn.BCEWithLogitsLoss() loss for the adversarial, discriminator, real and fake loss?

**Comment by LZ-CH** [comment]
I carefully looked at the training process of the source code, and it is exactly as you said. The reason I did the discriminator training at the beginning was that, in my experience, discriminator pretraining makes the discrimination loss more stable. But I haven't tried training with the discriminator loss, you can try to adjust the discriminator training start time later.

**Comment by caaxgave** [comment]
I have another question. Is there any reason you initialize weight in that way? I cannot find the way they initialize them in the original implementation. Thank you in advance.

**Comment by LZ-CH** [comment]
In the supplementary material of the original paper, it is mentioned in the Additional Training Details chapter: We use He et al.’s method [15] to initialize the weights of our encoder and decoder conv layers, while the bias terms are initialized to zero.

[15] Kaiming He, Xiangyu Zhang, Shaoqing Ren, and Jian Sun. Delving deep into rectifiers: Surpassing human-level performance on ImageNet classification. In ICCV, 2015. 3
